picard/formats/__init__.py

```diff
@@ -38,6 +38,23 @@ def supported_formats():
         formats.append((format.EXTENSIONS, format.NAME))
     return formats
 
+def _guess_format(filename, options):
+    """Select the best matching file type amongst supported formats."""
+    fileobj = file(filename, "rb")
+    results = []
+    try:
+        header = fileobj.read(128)
+        options = [option for option in options if not option in (OggAudioFile, OggVideoFile) and option._File is not None]
+        results = [
+            (option._File.score(filename, fileobj, header), option.__name__, option)
+            for option in options]
+    finally:
```

Comment:
Is there a reason to use `try/finally` here instead of `with open(filename, "rb") as fileobj` ?